R/findSharedDrive.R

```diff
@@ -0,0 +1,100 @@
+# Project Name : epiuf
+# Script Name  : findSharedDrive.R
+# GitHub repo  : STRAP-epiuf
+# Summary      : 
+# Date created : 2024-05-23
+# Author       : Lore Merdrignac
+# Date reviewed:
+# Reviewed by  : 
+
+# Description -------------------------------------------------------------
+
+# Changes Log -------------------------------------------------------------
+
+
+# START of SCRIPT  --------------------------------------------------------
+
+
+#' findSharedDrive
+#' 
+#' Quick way to find the path to the root Shared Drive (e.g., "G:/Shared drive"). 
+#' It will identify the letter associated to the Drive (e.g., D:, G:, etc) 
+#' and then the name of the root folder (i.e., "Shared drive" or "Drive partagé").
+#' Indeed, depending on the user language, it could be labelled in english or French.
+#' Only these tow languages are implemented so far.
+#' 
+#' @param message boolean to print information messages or not
+#' @returns path to the shared drive
+#' @export
+#' @author STRAP team \email{strap@epiconcept.fr}
+#' @examples
+#' setPath("DRIVE", findSharedDrive())
+#' getPath("DRIVE")
+#' 
+
+findSharedDrive <- function(message = FALSE) {
+  
+  # Get the list of drives
+  os <- Sys.info()["sysname"]
+  
+  if (os == 'Windows') {
+    
+    # Windows command to get drives
+    drives <- system("wmic logicaldisk get caption,description", intern = TRUE)
+    # Remove the header line
+    drives <- drives[-1]
+    # Split the drive information into columns
+    drives_data <- strsplit(drives, "\\s+")
+    drives_data <- lapply(drives_data, function(x) x[x != ""])
+    drives_data <- do.call(rbind, drives_data)
+    # Extract the drive letters and names
+    drive_letters <- drives_data[, 1]
+    
+  } else if (os == 'Darwin') {
+    
+    # macOS command to get drives
+    drives <- system("df -h | grep '^/dev/' | awk '{print $1}'", intern = TRUE)
+    # Extract drive letters
+    drive_letters <- drives
+    
+  } else if (os == 'Linux') {
+    
+    # Linux command to get drives
+    drives <- system("lsblk -o NAME,MOUNTPOINT | grep '^sd' | awk '{print $1}'", intern = TRUE)
+    # Extract drive letters
+    drive_letters <- drives
+    
+  } else {
+    stop("Unsupported operating system")
+  }
+  
+
+  # Initialising the result to return
+  result <- NULL
+  
+  # Searching among the drives if Shared drive or Drive partag\u00E9s exist and store it
+  for (i in seq_along(drive_letters)) {
```

Comment:
Suggested simplified script to test each language, give message if not drive found, and also permit new languages to be found more easily. 
`# List all laguages to search for:
shared_drive_name <- c("Shared drives", "Drive partag\u00E9s")
# Searching among the drives if Shared drive or Drive partag\u00E9s exist and store it
for (i in seq_along(drive_letters)) {
  # Set parameters for the while loop:
  test_file <- FALSE
  y <- 1
  # If first, the path has not been found or the limit has been reached then carry on language option list
   while(!test_file&y<=length(shared_drive_name)){   
     # Add 'Shared drives' option to the drive letter and list files
    shared_drives_path <- file.path(drive_letters[i], shared_drive_name[y])
    # Check if drive exists
    test_file <- dir.exists(shared_drives_path)
  
  if (test_file) { # if exists, then set as result
    result <- shared_drives_path
    if (message == TRUE) message(paste("Shared drive found here:", result))
  } 
    y <- y+1 # roll over ticker
  }
}

if(is.null(result)){message("No shared drive identified")}else{
  return(result)
}`


Idea to solve problem of not knowing the names of shared drive in many languages : add option to manually specify the name of the shared drive

Reply:
May be the local config is the simplest solution ?